### DISCUSSION TITLE
플레이어의 키를 알수 잇는 방법이 있을까요 ?

### DISCUSSION THREAD

**Discussion by mhkim0720** [naverz/zepeto-studio-kor#118] [5. Q&A]
구현 하다 보니 플레이어의 키(Height)값이 필요한데 CharacterController 의 Height 값이 캐릭터 키와 상관없이 1.2 로 나오는거 같아서요. 
엄청 작은 캐릭터의 키를 구해오려면 어떻게 해야 하나요? Zepeto Context Scale 값이 보이긴 하는데 무슨일인지 
유니티 코딩하듯이 참조 하려고 해도 참조가 안되고 계속 1로 나옵니다.  
답변 부탁드려요 ~

**Comment by moondory77-naverz** [comment]
안녕하세요.
현재 로드되는 캐릭터의 키(height) 프로퍼티 값을 
캐릭터 컨트롤러의 모듈에서 직접적으로 제공하고 있지는 않습니다.

좋은 의견 감사하며, 다음 업데이트에 해당 프로퍼티 정보를 알 수 있는 방법을 
제공드릴 수 있도록 하겠습니다.

감사합니다.

**Reply by mhkim0720** [reply]
안녕하세요 

답변 감사합니다. 

키가 필요한 이유가 앉기를 제대로 구현하기 위해서 인데 그렇다면 프로퍼티 정보를 지원해주시기 전까지는 
구현이 불가능 한 부분이겠네요? 다른 우회 방법이 없을까요 ?

**Reply by x4cf** [reply]
생성된 캐릭터의 목 모델의 위치정보를 가져오는건 어떨까요?

<img width="387" alt="스크린샷 2021-12-24 오전 11 12 03" src="https://user-images.githubusercontent.com/90679926/147307296-a0901f59-7f0c-4474-82f1-61b88d1c04e3.png">

**Reply by mhkim0720** [reply]
생각 해봤던건데... 
확인해보시면 아시겠지만  캐릭터 키 모델 별로 Zepeto Context- body 하위 하이어라키 구조가 달라서 일괄적으로 목의 위치를 가져오는건 방법이 안될 거 같습니다.  어떤 모델은 어떤 게 있고 없고가 다르더라고요 

모든 캐릭터 모델에 neck_scale 이라는 오브젝트가 있다는게 확인이 되면 사용할 수 도 있기는 하겠지만 일일이 
다 확인 해보기가 .... 캐릭터 키를 다 바꿔보며 .... 하.. 공식 답변으로 neck_scale 객체가 무조건 
모델 파일에 있다고 해주시면 사용 해볼수는 있을 거 같아요

**Reply by x4cf** [reply]
그렇군요...

**Reply by mhkim0720** [reply]
추가적으로 플레이어의 각 소켓 ( head , hips, Know..  ) 등의 값을 읽어서 그 위치에 프리팹을 붙여 봤는데 

키에 맞는 위치가 안나오네요 

예를 들어 난쟁이 캐릭의  hips 는 프리팹이 머리에 붙고 보통캐릭은 엉덩이에 붙아요.  정상적인 값이 안 나오는 걸로 보이네요 

GG 입니다.

**Reply by x4cf** [reply]
추후에 제공한다고 하니 다른기능부터 만들어야겠네요, 그래도 모두 확인해보시고... 굿입니다!

**Reply by x4cf** [reply]
카메라가 헤드어퍼소켓을 기준으로 움직이는것같은데 이건 캐릭터마나 머리의 중앙으로 생성되는것으로 보입니다

<img width="264" alt="스크린샷 2021-12-24 오후 4 39 41" src="https://user-images.githubusercontent.com/90679926/147330351-84f82b72-06a1-44a0-9196-1ebaeaf2a62d.png">
